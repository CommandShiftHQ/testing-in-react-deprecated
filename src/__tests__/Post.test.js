import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Post from "../components/Post";

describe("Post", () => {
  const validProps = {
    postData: {
      author: "test author",
      body: "test body",
      date: "test date",
      isPublished: true,
      tags: ["test tag1", "test tag2", "test tag3"],
      title: "test title",
    },
    handleUpvote: jest.fn(),
  };

  test("renders as expected", () => {
    const testRenderer = renderer
      .create(
        <Post
          postData={validProps.postData}
          handleUpvote={validProps.handleUpvote}
        />
      )
      .toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  test("Post author renders correctly", () => {
    render(
      <Post
        postData={validProps.postData}
        handleUpvote={validProps.handleUpvote}
      />
    );

    expect(screen.getByText("Author: test author")).toBeInTheDocument();
  });
});