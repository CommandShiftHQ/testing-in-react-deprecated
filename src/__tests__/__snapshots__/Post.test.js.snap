// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders as expected 1`] = `
<div
  className="post"
>
  <div
    className="post-heading"
  >
    <h2>
      test title
    </h2>
    <p>
      test body
    </p>
  </div>
  <div
    className="post-counter"
  >
    <span>
      Upvotes: 
      0
    </span>
    <button
      onClick={[Function]}
      type="button"
      value="test title"
    >
      Upvote this
    </button>
  </div>
  <div
    className="post-author"
  >
    Author: 
    test author
  </div>
  <div
    className="post-date"
  >
    Published: 
    test date
  </div>
  <h3>
    Tags:
  </h3>
  <ul>
    <li>
      test tag1
    </li>
    <li>
      test tag2
    </li>
    <li>
      test tag3
    </li>
  </ul>
</div>
`;
